#!/usr/bin/env python
# agent pose prediction configuration

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, bool_t, str_t

gen = ParameterGenerator()

gen.add("velobs_mul", double_t, 0,
        "multiplier for agent velocities for velocity-obstacle calculation", 1.0, 0.001, 10.0)
gen.add("velobs_min_rad", double_t, 0,
        "minimum radius for velcoity-obstacle calculation", 0.25, 0.0, 10)
gen.add("velobs_max_rad", double_t, 0,
        "maxium radius for volocity-obstacle calculation", 0.75, 0.0, 10)
gen.add("velobs_max_rad_time", double_t, 0,
        "time for maximum radius for velocity-obstacle calculation", 4.0, 0.0, 60.0)
gen.add("velobs_use_ang", bool_t, 0,
        "whether to use angular velocity for velocity-obstacle calculation", True)

exit(gen.generate("agent_path_prediction",
     "agent_path_prediction", "AgentPathPrediction"))
